src/Compilers/CSharp/Test/Semantic/Semantics/GenericConstraintsTests.cs

```diff
@@ -0,0 +1,549 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.Semantics
+{
+    public class GenericConstraintsTests : CompilingTestBase
+    {
+        [Fact]
+        public void EnumConstraint_Compilation_Alone()
+        {
+            CreateStandardCompilation(@"
+public class Test<T> where T : System.Enum
+{
+}
+public enum E1
+{
+    A
+}
+public class Test2
+{
+    public void M()
+    {
+        var a = new Test<E1>();             // enum
+        var b = new Test<int>();            // value type
+        var c = new Test<string>();         // reference type
+        var d = new Test<System.Enum>();    // Enum type
```

Comment:
Another case that may be useful to try: generic type U which is itself constrained to `System.Enum`.